### PR TITLE
LIVY-316. NPE when starting kinit thread

### PR DESCRIPTION
**Pull request by zjffdu**
Looks like it caused by #285, Fix it in this PR.

### CODE REVIEW

File: server/src/main/scala/com/cloudera/livy/server/LivyServer.scala [hunk 1]
```diff
@@ -101,7 +101,7 @@ class LivyServer extends Logging {
       )
       val launch_keytab = livyConf.get(LAUNCH_KERBEROS_KEYTAB)
       val launch_principal = SecurityUtil.getServerPrincipal(
-        livyConf.get(LAUNCH_KERBEROS_PRINCIPAL), server.host)
+        livyConf.get(LAUNCH_KERBEROS_PRINCIPAL), host)
```
Comment (jerryshao) [inline server/src/main/scala/com/cloudera/livy/server/LivyServer.scala]: AFAIK kerberos requires FQDN, and here host is "0.0.0.0" by default, should we use a hostname instead?

Comment (zjffdu) [inline server/src/main/scala/com/cloudera/livy/server/LivyServer.scala]: `SecurityUtil` will do this

Comment (jerryshao) [inline server/src/main/scala/com/cloudera/livy/server/LivyServer.scala]: Ohh, I see, just noticed this. Looks like this is introduced in my previous patch.

**Comment by tc0312** [comment]
We really need some test coverage on Kerberos.

**Comment by jerryshao** [comment]
Let me spend some time on Hadoop's MiniKdc, to see if we could incorporate this to Livy's integration test.

**Pull request merged**